python/cudf/cudf/dataframe/multiindex.py

```diff
@@ -555,18 +556,23 @@ def to_pandas(self):
         pandas_codes = []
         for code in self.codes.columns:
             pandas_codes.append(self.codes[code].to_array())
+        pandas_levels = [
+            level.to_pandas() if hasattr(level, "to_pandas") else level
+            for level in self.levels
```

Comment:
Do we support multiple object types for levels?

Reply:
This is mostly accurate: Levels produced by `groupby` or any other `_source_data` effect are `Series`. Levels that are specified by the user are simple lists.

Reply:
Don't we cast those to `Series` in MultiIndex object creation?

Reply:
Yes, I've cleaned this up.